// @wdio/allure-reporter/cjs/common/api.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.addArgument = exports.addStep = exports.endStep = exports.startStep = exports.addAttachment = exports.addDescription = exports.addEnvironment = exports.addStory = exports.addTestId = exports.addIssue = exports.addSeverity = exports.addLabel = exports.addFeature = void 0;
const constants_js_1 = require("../constants.js");
/**
 * Call reporter
 * @param {string} event  - event name
 * @param {Object} msg - event payload
 * @private
 */
const tellReporter = (event, msg = {}) => {
    process.emit(event, msg);
};
/**
 * Assign feature to test
 * @name addFeature
 * @param {(string)} featureName - feature name or an array of names
 */
function addFeature(featureName) {
    tellReporter(constants_js_1.events.addFeature, { featureName });
}
exports.addFeature = addFeature;
/**
 * Assign label to test
 * @name addLabel
 * @param {string} name - label name
 * @param {string} value - label value
 */
function addLabel(name, value) {
    tellReporter(constants_js_1.events.addLabel, { name, value });
}
exports.addLabel = addLabel;
/**
 * Assign severity to test
 * @name addSeverity
 * @param {string} severity - severity value
 */
function addSeverity(severity) {
    tellReporter(constants_js_1.events.addSeverity, { severity });
}
exports.addSeverity = addSeverity;
/**
 * Assign issue id to test
 * @name addIssue
 * @param {string} issue - issue id value
 */
function addIssue(issue) {
    tellReporter(constants_js_1.events.addIssue, { issue });
}
exports.addIssue = addIssue;
/**
 * Assign TMS test id to test
 * @name addTestId
 * @param {string} testId - test id value
 */
function addTestId(testId) {
    tellReporter(constants_js_1.events.addTestId, { testId });
}
exports.addTestId = addTestId;
/**
 * Assign story to test
 * @name addStory
 * @param {string} storyName - story name for test
 */
function addStory(storyName) {
    tellReporter(constants_js_1.events.addStory, { storyName });
}
exports.addStory = addStory;
/**
 * Add environment value
 * @name addEnvironment
 * @param {string} name - environment name
 * @param {string} value - environment value
 */
function addEnvironment(name, value) {
    tellReporter(constants_js_1.events.addEnvironment, { name, value });
}
exports.addEnvironment = addEnvironment;
/**
 * Assign test description to test
 * @name addDescription
 * @param {string} description - description for test
 * @param {string} descriptionType - description type 'text'\'html'\'markdown'
 */
function addDescription(description, descriptionType) {
    tellReporter(constants_js_1.events.addDescription, { description, descriptionType });
}
exports.addDescription = addDescription;
/**
 * Add attachment
 * @name addAttachment
 * @param {string} name         - attachment file name
 * @param {*} content           - attachment content
 * @param {string=} mimeType    - attachment mime type
 */
function addAttachment(name, content, type) {
    if (!type) {
        type = content instanceof Buffer ? 'image/png' : typeof content === 'string' ? 'text/plain' : 'application/json';
    }
    tellReporter(constants_js_1.events.addAttachment, { name, content, type });
}
exports.addAttachment = addAttachment;
/**
 * Start allure step
 * @name startStep
 * @param {string} title - step name in report
 */
function startStep(title) {
    tellReporter(constants_js_1.events.startStep, title);
}
exports.startStep = startStep;
/**
 * End current allure step
 * @name endStep
 * @param {StepStatus} [status='passed'] - step status
 */
function endStep(status = 'passed') {
    if (!Object.values(constants_js_1.stepStatuses).includes(status)) {
        throw new Error(`Step status must be ${Object.values(constants_js_1.stepStatuses).join(' or ')}. You tried to set "${status}"`);
    }
    tellReporter(constants_js_1.events.endStep, status);
}
exports.endStep = endStep;
/**
 * Create allure step
 * @name addStep
 * @param {string} title - step name in report
 * @param {Object} [attachmentObject={}] - attachment for step
 * @param {string} attachmentObject.content - attachment content
 * @param {string} [attachmentObject.name='attachment'] - attachment name
 * @param {string} [attachmentObject.type='text/plain'] - attachment type
 * @param {string} [status='passed'] - step status
 */
function addStep(title, { content, name = 'attachment', type = 'text/plain' } = {}, status = 'passed') {
    if (!Object.values(constants_js_1.stepStatuses).includes(status)) {
        throw new Error(`Step status must be ${Object.values(constants_js_1.stepStatuses).join(' or ')}. You tried to set "${status}"`);
    }
    const step = content ? { title, attachment: { content, name, type }, status } : { title, status };
    tellReporter(constants_js_1.events.addStep, { step });
}
exports.addStep = addStep;
/**
 * Add additional argument to test
 * @name addArgument
 * @param {string} name - argument name
 * @param {string} value - argument value
 */
function addArgument(name, value) {
    tellReporter(constants_js_1.events.addArgument, { name, value });
}
exports.addArgument = addArgument;
exports.default = {
    addFeature, addLabel, addSeverity, addIssue, addTestId, addStory, addEnvironment,
    addDescription, addAttachment, startStep, endStep, addStep, addArgument
};
